tests/platform_tests/api/test_chassis_fans.py

```diff
@@ -252,13 +239,14 @@ def test_set_fans_speed(self, duthosts, enum_rand_one_per_hwsku_hostname, localh
                 target_speed = random.randint(speed_minimum, speed_maximum)
 
             speed = fan.get_speed(platform_api_conn, i)
-            speed_tol = fan.get_speed_tolerance(platform_api_conn, i)
 
             speed_set = fan.set_speed(platform_api_conn, i, target_speed)       # noqa F841
-            time.sleep(5)
+            time.sleep(self.get_fan_facts(duthost, i, 5, "speed", "delay"))
 
             act_speed = fan.get_speed(platform_api_conn, i)
-            self.expect(abs(act_speed - target_speed) <= speed_tol,
+            under_speed = fan.is_under_speed(platform_api_conn, i)
```

Comment:
Can we also check over_speed > under_speed?

Reply:
`is_under_speed` and `is_over_speed` return boolean values; a comparison wouldn't make sense here.

Reply:
Got it. thanks